dashboard/lti_new.py

```diff
@@ -75,6 +79,48 @@ def get_jwks(request):
     return JsonResponse(config.get_jwks())
 
 
+def recursive_format(args, **kwargs):
+    """
+    Recursively apply `string.format()` to all strings in a data structure.
+
+    This is intended to be used on a data structure that may contain
+    format strings just before it is passed to `json.dump()` or `dumps()`.
+
+    :param args: data structure; mostly dictionaries and lists
+    :param kwargs: values to be used in `string.format()`
+    :return: a new data structure with formatted strings
+    """
+    if isinstance(args, Mapping):
+        return {key: recursive_format(value, **kwargs)
+                for (key, value) in args.items()}
+    elif isinstance(args, MutableSequence):
+        return [recursive_format(value, **kwargs)
+                for value in args]
+    else:
+        return args.format(**kwargs)
+
+
+def generate_config_json(request: HttpRequest) -> Union[
+    JsonResponse, LTIError]:
+    config = get_tool_conf()
+    if not check_if_success_getting_tool_config(config):
+        return LTIError(config).response_json()
+
+    params = {
+        'timestamp': datetime.now().isoformat(),
+        'host': request.get_host(),
+        'base_url': urllib.parse.urlunsplit(
+            [request.scheme, request.get_host()] + 3 * ['']),
+        'url_login': reverse('login'),
+        'url_launch': reverse('launch'),
+        'url_get_jwks': reverse('get_jwks'),
```

Comment:
I like the way you were autofilling this rather me changing it manual in config

Reply:
Thanks.